### PR TITLE
feat: add login functionality

### LINKED ISSUE #3: feat: add db logic for User Authentication.

### PR DESCRIPTION
**Pull request by Mustafa2856**
The login functionality along with database support has been implemented

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -1,7 +1,13 @@
-import requests
-from flask import Flask, request, jsonify
-
+from flask import Flask, request
+from flask_sqlalchemy import SQLAlchemy
+import os
```
Comment (GrayFlash) [inline app.py:3]: we can add a blank line after import statements end, before declaring variables.

File: app.py [hunk 2]
```diff
@@ -1,7 +1,13 @@
...
 app = Flask(__name__)
+s = os.environ.get("DATABASE_URL")
```
Comment (GrayFlash) [inline app.py]: can use a more descriptive variable name

File: db/tables/user.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+from app import db
+
+class User(db.Model):
```
Comment (GrayFlash) [inline db/tables/user.py]: If this table is only for Authentication, let's rename it to `AuthTable`, else if we are gonna use the same table for other userdata rename it to `UserTable`.

File: db/tables/user.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    def check_password(self,password):
+        return self.password == password
+    
+    @classmethod
+    def login_user(cls,username,password):
+        user = cls.query.filter_by(username=username).first()
+        if user == None:
+            return None
+        if user.check_password(password):
+            return user
```
Comment (GrayFlash) [inline db/tables/user.py]: leave an empty line at the end of every file.

**Review by GrayFlash — CHANGES_REQUESTED** [review]

**Comment by Mustafa2856** [comment]
@GrayFlash Changes as requested have been implemented

File: app.py [hunk 5]
```diff
@@ -1,10 +1,11 @@
 from flask import Flask, request
 from flask_sqlalchemy import SQLAlchemy
 import os
+
 app = Flask(__name__)
-s = os.environ.get("DATABASE_URL")
-s = 'postgresql+psycopg2:/' + s[s.find('/')+1:]
-app.config['SQLALCHEMY_DATABASE_URI'] = s
+database_url = os.environ.get("DATABASE_URL")
```
Comment (GrayFlash) [inline app.py]: use `db_url` instead

**Review by GrayFlash — CHANGES_REQUESTED** [review]

**Comment by GrayFlash** [comment]
@Mustafa2856 please write commit message as `refactor: <message>`

**Review by GrayFlash — APPROVED** [review]

**Pull request merged**